src/core/qgsvectorlayerjoininfo.h

```diff
@@ -131,6 +131,36 @@ class CORE_EXPORT QgsVectorLayerJoinInfo
      */
     QgsFeature extractJoinedFeature( const QgsFeature &feature ) const;
 
+    /**
+     * Sets a list of fields to ignore whatever happens.
+     *
+     * \since QGIS 3.0
+     */
+    void setJoinFieldNamesBlackList( const QStringList &blackList ) { mBlackList = blackList; }
```

Comment:
What's this blacklist used for?
Aren't joins able to ignore certain fields already? I was just thinking "never say never" ;)

Reply:
Yes it's currently possible to ignore certain fields with the subset mechanism. However, it's completely static. 

If you have a joined layer with `fieldA` and `fieldB`, you may define a subset to only see `fieldA` from the target layer and ignore `fieldB`.

Then, if you add some fields in the joined layer, like `fieldC`, the target layer won't see it as `fieldC` is not in the subset.

However, it's constraining in our case because we add on the fly new columns in the joined layer (the auxiliary layer) when a new property has to be managed. 

So I added a blacklist mechanism which is fully dynamic. You just have to give some field to ignore and then, even if some fields are added in the joined layer, then the target layer will see them.

Reply:
Will it be possible for a user to unblacklist fields? Or is this something we really don't want someone to do?

Reply:
I asked myself the same question. As I wasn't sure, the blacklist option is not available in ui for users...